src/aiodynamo/credentials.py

```diff
@@ -289,6 +293,81 @@ def and_then(thing: Optional[T], mapper: Callable[[T], U]) -> Optional[U]:
     return thing
 
 
+# https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRoleWithWebIdentity.html
+@dataclass
+class AssumeRoleWithWebIdentityCredentials(MetadataCredentials):
+    ACTION = "AssumeRoleWithWebIdentity"
+    VERSION = "2011-06-15"
+
+    timeout: Timeout = 2
+    max_attempts: int = 3
+
+    base_url: URL = URL("https://sts.amazonaws.com/")
+
+    async def fetch_metadata(self, http: HttpImplementation) -> Metadata:
+        config: dict = self._load_config()
```

Comment:
I'd prefer this not to be an un-typed dict (`dict` is a very broad type, but you know what keys/values are gonna be there anyway). Either use a `TypedDict`, `dataclass` or since there's always three values from what I can see, a `tuple[str, str, str]`